scheduler/src/cook/mesos/util.clj

```diff
@@ -438,3 +439,32 @@
   [instance]
   (some #{(:instance/status instance)} #{:instance.status/running
                                          :instance.status/unknown}))
+
+(defn close-when-ch!
+  "When the value passed in is a channel, close it. Otherwise, do nothing"
+  [maybe-ch]
+  (try
+    (async/close! maybe-ch)
+    (catch Exception e)))
+
+(defn chime-at-ch
```

Comment:
In many ways, this is just an actor that is reading messages from its queue (`ch`) and processing them (by calling `f`). I'm not sure we want to have `chime` in the name of this function despite its origin.